messages.go

```diff
@@ -98,6 +99,20 @@ type Event struct {
 	Content    string      `json:"content"`
 }
 
+func (e *Event) String() string {
```

Comment:
Go structs already have a natural format for printing. See [this Go playground as an example](https://go.dev/play/p/1R6_QVQFgMv). See also the [Printing](https://pkg.go.dev/fmt#hdr-Printing) section in the std library's `fmt` package.

What advantage does this custom implementation bring over the `fmt` package's implementation? If you want to conform the types to the `Stringer` interface, perhaps printing the struct into a string is better:

```suggestion
func (e *Event) String() string {
    return fmt.Sprintf("%+v", e)
}
```

Reply:
I tried to address the bytes being printed out to the stdout, as Jiri commented above. If the data contains only ASCII characters, they can be represented as text instead of raw byte values.